src/WebSocketBuilder.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Net.WebSockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s
+{
+    /// <summary>
+    /// The <see cref="WebSocketBuilder"/> creates a new <see cref="WebSocket"/> object which connects to a remote WebSocket.
+    /// </summary>
+    /// <remarks>
+    /// By default, this uses the .NET <see cref="ClientWebSocket"/> class, but you can inherit from this class and change it to
+    /// use any class which inherits from <see cref="WebSocket"/>, should you want to use a third party framework or mock the requests.
+    /// </remarks>
+    public class WebSocketBuilder
```

Comment:
This class doesn't seem to add much imho.  Why not just inline the various calls in the class in the code above...

Reply:
The calls are not inlined because the unit tests use a mock `WebSocketBuilder` which traces all the parameters which have been passed and returns a dummy `WebSocket`.

I used a Builder-pattern becuase the code also uses the `UriBuilder` class.

I can change it a function/property like `GetWebSocket(Dictionary<string,string> headers, Collection<X509Certificate2> clientCertificates, Uri uri)`, though, if that's what you'd prefer. As long as the unit tests can intercept the call it's fine with ime.

Reply:
Injectable for testing is fair...